### PR TITLE
[node] Update types

### PR DESCRIPTION
**Pull request by Semigradsky**
https://github.com/nodejs/node/releases/tag/v18.14.0

- http: join authorization headers

https://github.com/nodejs/node/releases/tag/v18.15.0

- test_runner: add reporters
- vm: expose cachedDataRejected for vm.compileFunction
- v8: support gc profile
- process: add constrainedMemory API

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

File: types/node/ts4.8/test.d.ts [hunk 1]
```diff
@@ -214,25 +248,35 @@ declare module 'node:test' {
...
-         * The test duration.
+         * Additional execution metadata.
          */
-        duration: number;
+        details: {
+            /**
+             * The duration of the test in milliseconds.
+             */
+            duration: number;
+        }
```
Comment (peterblazejewicz) [inline types/node/ts4.8/test.d.ts:268]:
```suggestion
        };
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@Semigradsky thanks!

File: types/node/vm.d.ts [hunk 2]
```diff
@@ -446,7 +446,9 @@ declare module 'vm' {
      * @param code The body of the function to compile.
      * @param params An array of strings containing all parameters for the function.
      */
-    function compileFunction(code: string, params?: ReadonlyArray<string>, options?: CompileFunctionOptions): Function;
+    function compileFunction(code: string, params?: ReadonlyArray<string>, options?: CompileFunctionOptions): Function & {
+        cachedDataRejected: Script['cachedDataRejected'];
```
[the remark this answers is not in the document]
Comment (Semigradsky) [reply types/node/vm.d.ts:450]: Yes. Also I see that function can has `cachedDataProduced` and `cachedData` properties.
https://github.com/nodejs/node/pull/46320/files
I will add

[the remark this answers is not in the document]
Comment (Semigradsky) [reply types/node/vm.d.ts:450]: I will create separate PR for this.

**Comment by Semigradsky** [comment]
Ready to merge

**Pull request merged**